helix-core/src/main/java/org/apache/helix/store/zk/ZkHelixPropertyStore.java

```diff
@@ -47,4 +48,9 @@ public ZkHelixPropertyStore(String zkAddress, ZkSerializer serializer, String ch
       ZkBaseDataAccessor.ZkClientType zkClientType) {
     super(zkAddress, serializer, chrootPath, null, null, MONITOR_TYPE, chrootPath, zkClientType);
   }
+
+  // test only
+  protected HelixZkClient getTestZkClient() {
+    return getZkClient();
+  }
```

Comment:
Exposing the internal HelixZkClient is not desirable

Reply:
Java has not way to access otherwise private members in the class. In this case, we have to expire the session inside this class. This is a common problem I did not see addressed in our code base. Unless you guys can identify another pattern, I guess we have to leave it this way.

Reply:
We have another thread of this discussion in this pull. Close this one